src/main/java/com/idrsolutions/JDeliImageReader.java

```diff
@@ -113,14 +115,15 @@ public int getNumImages(final boolean allowSearch) {
      */
     @Override
     public int getWidth(final int imageIndex) throws IOException {
-        final int w;
+        int w;
         if (delegate == null) {
+            getByteArray();
             try {
-                readMetadata();
-            } catch (final Exception ex) {
-                ex.getStackTrace();
+                w = JDeli.readDimension(bytes).width;
```

Comment:
I personally not a fan of code like this which can return null and catches in exception.

I would recommend
Rectangle dimensions = JDeli.readDimension(bytes);
if(dimensions != null){
w = dimensions.width;
}